src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -179,47 +183,123 @@ class ExternalDatabaseManagementService(
         return columnNamesByTableName
     }
 
+    /*UPDATE*/
+    fun updateOrganizationExternalDatabaseTable(orgId: UUID, tableFqnToId: Pair<String, UUID>, update: MetadataUpdate) {
+        update.name.ifPresent {
+            val newTableFqn = FullQualifiedName(orgId.toString(), it)
+            val oldTableName = getNameFromFqnString(tableFqnToId.first)
+            val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+            assemblerConnectionManager.connect(dbName).let { dataSource ->
+                dataSource.connection.createStatement().use { stmt ->
```

Comment:
dataSource.connection checks out a connection from the pool -- you need to close it.

conversely you don't need to close statements for connections you're closing, so you can replace this with
dataSource.connection.use { conn -> 
  stmt = conn.createStatement()
   stmt.execute(...)
}